lib/puppet/agent.rb

```diff
@@ -60,6 +61,23 @@ def run(client_options = {})
               end
             end
           rescue Puppet::LockError
+            if client_options[:waitforlock]
+              wait_for_lock_deadline ||= Time.now.to_i + Puppet[:maxwaitforlock]
+
+              if Puppet[:maxwaitforlock] < 1
+                Puppet.notice _("Run of %{client_class} already in progress; skipping  (%{lockfile_path} exists, and waitforlock setting is set to 0)") % { client_class: client_class, lockfile_path: lockfile_path }
+                return
```

Comment:
Due to PUP-10366, all of these returns need to `return 1`, otherwise we end up returning `nil` implicitly which gets passed as `exit(nil)`